src/emqx_packet.erl

```diff
@@ -77,6 +82,19 @@ validate_properties(?PUBLISH, #{'Topic-Alias':= I})
     error(topic_alias_invalid);
 validate_properties(?PUBLISH, #{'Subscription-Identifier' := _I}) ->
     error(protocol_error);
+validate_properties(?PUBLISH, #{'Response-Topic' := ResponseTopic}) ->
+    case emqx_topic:wildcard(ResponseTopic) of
+        true ->
+            error(protocol_error);
```

Comment:
here and below

wouldn't it be more informative for troubleshooting if we raise the errors with extra info like:
`error({protocol_error, wildcard_not_allowed_as_response_topic})` ?

Reply:
The mqtt 5.0 specification did not specifed the `wildcard_not_allowed_as_response_topic` error. So I supposed it is unnecessary to add this information.

Reply:
Additional information should be included in User Properties or Reason String.

Reply:
`Reason String` is the  property of MQTT5.0 packet. Usually,  broker uses this property to give additional information to the Client.

Reply:
I mean do we provide a `Reason String` to the client in case of `protocol_error` ?
If not, wouldn't it be nice to provide ? (in another pull request perhaps).

Reply:
@emqplus said it is unnecessary to add reason string in case of `protocol_error`, we need to discuss it  before next milestone.